2-Pricing/src/App.tsx

```diff
@@ -0,0 +1,69 @@
+import { useState } from "react";
+
+import { Modal } from "./components/Modal";
+import { View } from "./components/View";
+import { Header } from "./components/Header";
+import { Subscriptions } from "./components/Subscriptions";
+
+import { Subscription, subscriptions } from "./constants";
+import { useLocalStorage } from "./hooks/useLocalStorage";
+import { CartIcon } from "./ui/icons";
+import { Cart as CartComponent } from "./components/Cart";
+
+export interface Cart {
+  alpha?: number;
+  bravo?: number;
+  charlie?: number;
+  extra?: number;
+}
+
+function App() {
+  const [inCart, setinCart] = useLocalStorage<Cart>("cart", {});
+  const [isMonthlyView, setIsMonthlyView] = useState(true);
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [isCartOpen, setIsCartOpen] = useState(false);
+  const [subscriptionCandidate, setSubscriptionCandidate] =
+    useState<Subscription>(subscriptions.bravo);
+
+  return (
+    <main className={`${isCartOpen || isModalVisible ? "no-scroll" : ""}`}>
+      <button
+        id="cart-icon"
+        className={`${isCartOpen ? "first-layer" : "second-layer"}`}
+        onClick={() => setIsCartOpen((prevIsCartOpen) => !prevIsCartOpen)}
+      >
+        <CartIcon />
+      </button>
+
+      <Header />
+
+      <View isMonthlyView={isMonthlyView} setIsMonthlyView={setIsMonthlyView} />
+
+      {isModalVisible && (
+        <Modal
+          setIsModalVisible={setIsModalVisible}
+          subscriptionCandidate={subscriptionCandidate}
+          inCart={inCart}
+          setInCart={setinCart}
+        />
+      )}
+
+      {isCartOpen && (
+        <CartComponent
+          isMonthlyView={isMonthlyView}
+          inCart={inCart}
+          setInCart={setinCart}
+          setIsCartOpen={setIsCartOpen}
+        />
+      )}
```

Comment:
I think these two should be outside of `<main>`, maybe we can make `App` return a Fragment with `main` and these two components.

Reply:
Do you think the `<button id="cart-icon"...>` also belong outside? I think it sould be in the same section of the document as the cart